### PR TITLE
Adding filterNotNull method.

### PR DESCRIPTION
**Pull request by terhesb**
In this change I implemented the filterNotNull method.

### CODE REVIEW

File: src/test/java/com/github/underscore/CollectionsTest.java [hunk 1]
```diff
@@ -631,6 +631,16 @@ public boolean test(Integer item) {
         assertEquals("[2, 4, 6]", resultObject.toString());
     }
 
+    @Test
+    public void filterNotNull() {
+        final List<Integer> result = U.filterNotNull(asList(1, null, 3, 4, Integer.getInteger(null)));
```
Comment (sputnikci) [inline src/test/java/com/github/underscore/CollectionsTest.java]: \[FindBugs\] WARNING: NP: Non\-virtual method call in com\.github\.underscore\.CollectionsTest\.filterNotNull\(\) passes null for nonnull parameter of Integer\.getInteger\(String\)

Comment (sputnikci) [inline src/test/java/com/github/underscore/CollectionsTest.java]: \[FindBugs\] WARNING: NP: Non\-virtual method call in com\.github\.underscore\.CollectionsTest\.filterNotNull\(\) passes null for nonnull parameter of Integer\.getInteger\(String\)

**Comment by javadev** [comment]
Hi, @terhesb !

Thank you for pull request with filterNotNull.

I think it will be better to implement static method nonNull like in java 8.

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-

and use it in filter `U.filter(iterable, U::nonNull).

We have U.compact method which removes nulls.

**Comment by terhesb** [comment]
Okay

**Pull request closed without merging**